README.md

```diff
@@ -420,6 +421,89 @@ status:
 - Since `error_msg_path` is specified, kubernetes-deploy will log the contents of `$.status.conditions[?(@.type == "Failed")].message`, which in this case is: `resource is failed`.
 
+### Deploy walkthrough
+
+Let's walk through what happens when you run the `deploy` task with [this directory of templates](https://github.com/Shopify/kubernetes-deploy/tree/master/test/fixtures/hello-cloud). You can see this for yourself by running the following command:
+
+```bash
+krane deploy my-namespace my-k8s-cluster -f test/fixtures/hello-cloud --render-erb
+```
+
+As soon as you run this, you'll start seeing some output being streamed to STDERR.
+
+#### Phase 1: Initializing deploy
+
+In this phase, we:
+
+- Perform basic validation to ensure we can proceed with the deploy. This includes checking if we can reach the context, if the context is valid, if the namespace exists within the context, and more. We try to validate as much as we can before trying to ship something because we want to avoid having an incomplete deploy in case of a failure (this is especially important because there's no rollback support).
+- List out all the resources we want to deploy (as described in the template files we used).
+- Render ERB templates and apply partials, if enabled (which is the case for this example). If enabled, we also perform basic validation on the parsed templates.
+
+#### Phase 2: Checking initial resource statuses
+
+In this phase, we check resource statuses. For each resource listed in the previous step, we check Kubernetes for their status; in the first deploy this might show a bunch of items as "Not Found", but for the deploy of a new version, this is an example of what it could look like:
+
+```
+Certificate/services-foo-tls     Exists
+Cloudsql/foo-production          Provisioned
+Deployment/jobs                  3 replicas, 3 updatedReplicas, 3 availableReplicas
+Deployment/web                   3 replicas, 3 updatedReplicas, 3 availableReplicas
+Ingress/web                      Created
+Memcached/foo-production         Healthy
+Pod/db-migrate-856359            Unknown
+Pod/upload-assets-856359         Unknown
+Redis/foo-production             Healthy
+Service/web                      Selects at least 1 pod
+```
+
+The next phase might be either "Predeploying priority resources" (if there's any) or "Deploying all resources". In this example we'll go through the former, as we do have predeployable resources.
+
+#### Phase 3: Predeploying priority resources
+
+This is the first phase that could modify the cluster.
+
+In this phase we predeploy certain types of resources (e.g. `ConfigMap`, `PersistentVolumeClaim`, `Secret`) to make sure the latest version will be available when resources that might consume them (e.g. `Deployment`) are deployed. This will not show up if the templates don't include resources of a type that would be relevant here.
```

Comment:
> This will not show up if the templates don't include resources of a type that would be relevant here.

Might be worth massaging this wording a bit, this line tripped me up.
Maybe something like:
"This phase will be skipped if the templates don't include any resources that would need to be predeployed." or something to that effect.